bitnami/kafka/templates/scripts-configmap.yaml

```diff
@@ -62,177 +55,300 @@ data:
+        read -r -a command <<< "$cmd"
+        for ((i = 1 ; i <= retries ; i+=1 )); do
+            "${command[@]}" && return_value=0 && break
+            sleep "$sleep_time"
+        done
+        return $return_value
+    }
+
+    replace_in_file() {
+        local filename="${1:?filename is required}"
+        local match_regex="${2:?match regex is required}"
+        local substitute_regex="${3:?substitute regex is required}"
+        local posix_regex=${4:-true}
+
+        local result
+
+        # We should avoid using 'sed in-place' substitutions
+        # 1) They are not compatible with files mounted from ConfigMap(s)
+        # 2) We found incompatibility issues with Debian10 and "in-place" substitutions
+        local -r del=$'\001' # Use a non-printable character as a 'sed' delimiter to avoid issues
+        if [[ $posix_regex = true ]]; then
+            result="$(sed -E "s${del}${match_regex}${del}${substitute_regex}${del}g" "$filename")"
         else
-            export KAFKA_CFG_NODE_ID="$(grep "node.id" "{{ .Values.logsDirs | splitList "," | first }}/meta.properties" | awk -F '=' '{print $2}')"
+            result="$(sed "s${del}${match_regex}${del}${substitute_regex}${del}g" "$filename")"
         fi
-    else
-        if [[ $KAFKA_CFG_PROCESS_ROLES == "" ]]; then
-            export KAFKA_CFG_BROKER_ID="$((ID + {{ .Values.minId }}))"
+        echo "$result" > "$filename"
+    }
+
+    kafka_conf_set() {
+        local file="${1:?missing file}"
+        local key="${2:?missing key}"
+        local value="${3:?missing value}"
+
+        # Check if the value was set before
+        if grep -q "^[#\\s]*$key\s*=.*" "$file"; then
+            # Update the existing key
+            replace_in_file "$file" "^[#\\s]*${key}\s*=.*" "${key}=${value}" false
         else
-            export KAFKA_CFG_NODE_ID="$((ID + {{ .Values.minId }}))"
+            # Add a new key
+            printf '\n%s=%s' "$key" "$value" >>"$file"
         fi
-    fi
+    }
 
-    if [[ $KAFKA_CFG_PROCESS_ROLES == *"controller"* && -z $KAFKA_CFG_CONTROLLER_QUORUM_VOTERS ]]; then
-        node_id={{ .Values.minId }}
-        pod_id=0
-        while :
-        do
-            VOTERS="${VOTERS}$node_id@{{  include "common.names.fullname" . }}-$pod_id.{{ include "common.names.fullname" . }}-headless.{{ $releaseNamespace }}.svc.{{ $clusterDomain }}:{{ .Values.service.ports.controller }}"
-            node_id=$(( $node_id + 1 ))
-            pod_id=$(( $pod_id + 1 ))
-            if [[ $pod_id -ge {{ .Values.replicaCount }} ]]; then
-                break
-            else
-                VOTERS="$VOTERS,"
-            fi
-        done
-        export KAFKA_CFG_CONTROLLER_QUORUM_VOTERS=$VOTERS
-    fi
-    {{- if eq .Values.brokerRackAssignment "aws-az" }}
-    export KAFKA_CFG_BROKER_RACK=$(curl "http://169.254.169.254/latest/meta-data/placement/availability-zone-id")
```

Comment:
Was this intentionally dropped? This is a feature that was introduced in PR #13308 that we were relying on.